cpp/include/cudf/ast/detail/operators.hpp

```diff
@@ -780,6 +789,51 @@ struct operator_functor<ast_operator::NOT, false> {
   }
 };
 
+// Type traits used to determine when a casting operation is valid.
+template <typename T, typename U, typename = void>
+struct can_cast : std::false_type {
+};
+
+template <typename T, typename U>
+struct can_cast<T, U, std::void_t<decltype(static_cast<U>(std::declval<T>()))>> : std::true_type {
```

Comment:
You can just use `std::is_convertible` instead.

Also, you can simplify these functors a bit with a common templated one:

```
template <typename To>
struct cast{
   static constexpr auto arity{1};
   template <typename From, CUDF_ENABLE_IF(std::is_convertible_v<From, To>) >
   To operator()(From f){
     return static_cast<T>(u);  
   }
};

template <> struct operator_functor<ast_operator::CAST_TO_INT64, false> : cast<int64_t> {};
template <> struct operator_functor<ast_operator::CAST_TO_UINT64, false> : cast<uint64_t> {};
template <> struct operator_functor<ast_operator::CAST_TO_FLOAT64, false> : cast<double> {};

```